cps/tasks/download.py

```diff
@@ -132,7 +129,7 @@ def run(self, worker_thread):
                     self.stat = STAT_FINISH_SUCCESS
                     log.info("Download task for %s completed successfully", self.media_url)
                 else:
-                    self.end_time = datetime.now()                    
+                    self.end_time = datetime.now()
```

Comment:
If end_time needs to be set on both branches (lines 128 and 132), do it just once above the if.

By the way, this class sets end_time in multiple places (e.g. line 26 and 72). If the goal is to record the end time after the task completes, can these lines be eliminated?

Note: in case end_time ends up being unset, CalibreTask.runtime() will fail. The finally should catch all exit paths from the while loop, except when self.media_url is unset (which skips the entire thing).

Reply:
The goal is to record the run time dynamically per 
https://github.com/iiab/calibre-web/blob/eb8302387b75714c9ab8addafac5df8af74e93c8/cps/services/worker.py#L232C5-L235C1
I'll remove the redundancy for 128 and 132, but if I remove 26 and 72, I will lose the dynamic run time messages. I'll move 72 to be more efficient.

Reply:
Done in https://github.com/iiab/calibre-web/pull/170/commits/de4fdceaf9ec0b91ce6afa33495f2fa93159b0c3